### ISSUE TITLE
Race condition on reusing Handle objects

### ISSUE THREAD

**Issue by lericson** [ajkxyz/jpeg4py#2]
Since both JPEG and Handle classes have delete methods, it is possible for Python to start destructing both Handle and JPEG classes at the same time. This leads to `release` being called on an object that is then added to the list of decompressors. I advise strongly against this kind of hack due to this.

My workaround for this at the moment is to simply not have a delete method on the JPEG class, but instead to add the decompressor object to the list when it's been used.

**Comment by lericson** [comment]
I use Python 3.6, by the way.

**Comment by ajkxyz** [comment]
Thank you for noticing this.

Such behaviour could happen when JPEG object is contained within an object which has circular references (http://bugs.python.org/issue23720). As the random destructor call order may speedup garbage collection, in my opinion this breaks object-oriented paradigm. I've looked at the python3.4+ garbage collection source code once, and it seems that it is not easy to fix it without rewriting entire engine.

So I've added a workaround as you have suggested.
